atlp-rwanda/ecommerce-app-champions-bn
# #26 Create REST API to update a cart

#### What does this PR do?

- update the product in a cart
- Calculate the new cart total.
#### Description of Task to be completed

 -  Create a PUT endpoint for updating quantity of items in the cart
- Create a feature to calculate new total
-  Create a feature to remove a product from the cart if it's quantity is zero
-  Create a feature for checking if a product is in the cart
-  Create a feature for checking if the quantity updated is in stock
-  Create a swagger documentation
#### How should this be manually tested?

- clone the repo
- use _npm install_ to install all packages
- Add all required environment variables following the .env.example provided
- use _npm run migrate:up_ to migrate migrations
- Run _npm run db:seeds_ to populate seeders in the database
- use npm run dev to start the server
- log in a buyer and copy the provided token on Swagger
- paste the copied token like this _Bearer {token}_
- add a product to the cart on Swagger
- input the productId in the provided id field in the update cart endpoint on Swagger
- input the quantity number you prefer 
- navigate to get all carts route on Swagger, all changes and calculations should be there
